pkg/apis/hive/v1/hiveconfig_types.go

```diff
@@ -7,6 +7,12 @@ import (
 
 // HiveConfigSpec defines the desired state of Hive
 type HiveConfigSpec struct {
+
+	// TargetNamespace is the namespace where the core Hive components should be run. Defaults to "hive". Will be
+	// created if it does not already exist.
+	// +optional
+	TargetNamespace string `json:"targetNamespace,omitempty"`
```

Comment:
Fix up the comments for the reference fields in HiveConfig so that they explicitly state that the resources should be in the target namespace. For example, the comment for the `AdditionalCertificateAuthoritiesSecretRef` field states that the secret should be in the hive namespace. Alternatively (or additionally), add a comment to the `TargetNamespace` field to state that all resources referenced in the HiveConfig should be in the target namespace.